### DISCUSSION TITLE
Pod failed due to insufficient resource. Can't it be queued?

### DISCUSSION THREAD

**Discussion by purin0604** [argoproj/argo-workflows#7565] [Q&A]
Hello, I am new to K8S and argoWF and having an issue with deployment of parallel pods.
In the workflow, I have 100 tasks and set the parallelism: 50 with resource requests: CPU 0.3.
Since I have 3 nodes of 4 cores available (12 CPU), I can only run 12/0.3=40 pods simultaneously even though parallelism is set to be 50.
I am aware of this issue, but I expected that if pods can't find requested resource from any node, argoWF would put the pod in the queue or pending until other pods exit and free up some capacity. However, those failed with the following msg
`"Pod Node didn't have enough resource: cpu, requested: 310, used: 3655, capacity: 3910"`
Is this what it's supposed to be? Can I put the pods in the queue instead?

**Comment by adammw** [comment]
If you're running Kubernetes v1.22 or later, this might be a result of https://github.com/kubernetes/kubernetes/issues/107679

**Reply by jiaflu** [reply]
Kubernetes v1.23.5 still meets the same issue.
